.changes/unreleased/Under the Hood-20241217-110536.yaml

```diff
@@ -0,0 +1,6 @@
+kind: Under the Hood
+body: Added new equals macro that handles null value checks in sql
+time: 2024-12-17T11:05:36.363421+02:00
+custom:
+    Author: adrianburusdbt
```

Comment:
@VersusFacit don't forget to add yourself as an author as well

Also not sure this is "under the hood" as we're making a meaningful improvement to how snapshots/tests work